client/server/pages/index.js

```diff
@@ -127,6 +128,8 @@ function getDefaultContext( request, response, entrypoint = 'entry-main' ) {
 	const reduxStore = createReduxStore( getCachedState( initialReducer, 'root' ) );
 	setStore( reduxStore, getCachedState );
 
+	const queryClient = await createQueryClient();
```

Comment:
The original version in #67573 used a shared `QueryClient` for all logged out requests, as opposed to a separate client for each request. That enabled caching responses.

One way to bring that back would be to share just the cache:
```js
const sharedCache = new QueryCache();

function handleRequest() {
  const queryClient = new QueryClient( { queryCache: sharedCache } );
  /* ... */
}
```
I'm not sure what other internal state does a `QueryClient` have, in addition to the cache.

Reply:
That looks like a better approach the original version.

I was hoping to restore the cache with the next PR. We should also consider only caching logged-out users.
For the next PR, I'd also like to include react-query hydration